tools/CreateNuget.ps1

```diff
@@ -68,17 +94,179 @@ function UpdateDependencyVersions([string]$Nuspec)
    Write-Host "Updated dependency versions of: $Nuspec"
 }
 
+
+# GetPackageRefFromProject
+#
+#     Helper function to get a specific PackageRef from a .csproj file
+#     Parses and returns a PackagRef object (defined above) that contains:
+#         Version - (version of the package)
+#         ConditionOperator - (the equality operator for a framework, == or !=)
+#         Framework - The framework this Packageref applies to: (net452, net457, netstandard2.0)
+#
+function GetPackageRefFromProject([string]$Id, [string]$CsProj, [string]$Framework)
+{
+    [xml]$xml = Get-Content -Path $CsProj -Encoding UTF8
+
+    $Xpath = "//Project/ItemGroup/PackageReference[@Include='" + $Id + "']"
+    $targetPackage = $xml | Select-XML -XPath $Xpath
+    $currentCondition = ""
+    $compOperator = ""
+    $currentFramework = ""
+    $currentVersion = ""
+
+    # There may be duplicates of the same package when there are different versions
+    # for different frameworks (i.e. ReactiveUI). Therefore if our search
+    # returns more than one node, then we take the one that matches 
+    # the Framework in its Condition
+
+    if ($targetPackage.Node.Count -gt 1)
+    {
+        foreach ($tn in $targetPackage.Node)
+        {
+            if ($tn.Condition -match $Framework )
+            {
+                $currentCondition = $tn.Condition
+                $currentVersion = $tn.Version
+            }
+        }
+    }
+    else
+    {
+        $currentCondition = $targetPackage.Node.Condition
+        $currentVersion = $targetPackage.Node.Version
+    }
+
+    if ($currentCondition -match "==")
+    {
+        $compOperator = "=="
+    }
+
+    if ($currentCondition -match "!=")
+    {
+        $compOperator = "!="
+    }
+
+    if ($currentCondition -match "net452")
+    {
+        $currentFramework = "net452"
+    }
+
+    if ($currentCondition -match "net472")
+    {
+        $currentFramework = "net472"
+    }

```

Comment:
If/when we update to 4.8, this will need to be changed. Is there a way to do this that's more independent, e.g. if `$currentCondition` _starts_ with "net" then copy its value into `$currentFramework`?

Reply:
We can, but it's kinda sticky because we have to assume the name convention will remain the same in the future. For example, in the nuspec file, one targetframework attribute is ".NETFramework4.5.2". But the corresponding node to get the PackageRefs in the csproj file would have "net452" in the conditional. So if I add robustness to loop through the nodes matching those two strings (as opposed to hard coding), I might test  with ".NETFramework4.8.0" in the nuspec and "net480" in the csproj file. That may be wrong. It may be ".NETFramework4.8" and "net48", and maybe something even different with netstandard frameworks. If I take that into consideration I might be adding even more lines to make the script robust (unless I assume a certain naming convention and just follow it). Seems adding the 4.8 nodes manually (or any new framework) once we officially support them is safer. 

Another alternative: 
With increased logic, this script is getting close to executable territory. I was thinking we could also do a console exe that could read the csproj files first and then just generate the nuspecs from scratch (or update the existing copies accordingly). We would still have to assume the string convention mentioned above, but any future changes would only have to be made in the executable (as opposed to the script and all of the nuspec files). 

Let me know what you think

Reply:
Yeah, makes sense. Let's keep it like this for the time being.